jest.setup.ts

```diff
@@ -20,3 +20,16 @@
 //
 
 import "@inrupt/jest-jsdom-polyfills";
+import { Request, Response, Headers, fetch } from "undici";
+
+if (
+  typeof globalThis.Response === "undefined" ||
+  typeof globalThis.Request === "undefined" ||
+  typeof globalThis.Headers === "undefined"
+) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  globalThis.Response = Response as any;
+  globalThis.Request = Request as any;
+  globalThis.Headers = Headers as any;
+  globalThis.fetch = Headers as any;
```

Comment:
This seems like it could be a utility in universal fetch?

Reply:
I was thinking about having it as part of `@inrupt/jest-jsdom-polyfills`. I should have been more explicit about this, but this PR (which should have been a draft) isn't meant to be merged directly, it's just part of an overall experiment I'm running for Node 18 support across our different libraries.